### DISCUSSION TITLE
Pass information from latex to python code

### DISCUSSION THREAD

**Discussion by dflvunoooooo** [leo-colisson/robust-externalize#26] [Q&A] [answered]
Is it possible to pass information from latex to python. I would like to set the plot width to the textwidth of latex.

**Comment by tobiasBora** [comment, marked as answer]
Yes, just create a new placeholder containing something like:
```
set placeholder eval={__WIDTH_LATEX__}{\lenToCmNoUnit{\textwidth}}
```
(this requires the very latest version from master, the older version had `\lenToCm` but you need to manually trim the `cm` at the end) You can choose using the optional argument the unit, for instance matplotlib expects inches:

MWE:
```
\documentclass{scrartcl} 

\usepackage{robust-externalize}

\begin{document}
\begin{figure}
  \centering
  \noindent\rule{.75\linewidth}{1mm}
  \begin{CacheMeCode}{python, set placeholder eval={__LINEWIDTH__}{\lenToCmNoUnit[in]{.75\linewidth}}}
import matplotlib.pyplot as plt
import matplotlib
from matplotlib.pyplot import figure
figure(figsize=(__LINEWIDTH__, __LINEWIDTH__))
year = [2014, 2015, 2016, 2017, 2018, 2019]
tutorial_count = [39, 117, 111, 110, 67, 29]
plt.plot(year, tutorial_count, color="#6c3376", linewidth=2)
plt.title("Simple plot")
plt.xlabel('Year')
plt.ylabel('Number of futurestud.io Tutorials')
print(get_filename_from_extension(".pgf"))
# https://stackoverflow.com/a/52587591/4987648
plt.savefig("__ROBEXT_OUTPUT_PDF__", bbox_inches="tight")   
\end{CacheMeCode}
  \caption{Test}%
  \label{py:test}
\end{figure}

\end{document}

% Local Variables:
% TeX-command-extra-options: "--shell-escape -halt-on-error"
% End:
```

**Reply by dflvunoooooo** [reply]
Thank you very much! Latex tells me, I am using version 2.2-unstable, is this the newest version? Because `lenToCmNoUnit` isn't recognised.

**Reply by tobiasBora** [reply]
The version you want is `2.5+unstable`. Just download the .sty from https://github.com/leo-colisson/robust-externalize/tree/master and copy it to your main folder.

**Reply by dflvunoooooo** [reply]
Perfect, it is working :) Thank you very much!
